src/Network/Wai/SAML2/Validation.hs

```diff
@@ -174,13 +174,33 @@ validateResponse cfg responseData = runExceptT $ do
     then pure ()
     else throwError InvalidSignature
 
+    assertion <- case responseEncryptedAssertion samlResponse of
+        Just encrypted -> decryptAssertion cfg encrypted
+        Nothing
+            | saml2RequireEncryptedAssertion cfg -> throwError $ InvalidResponse $ userError "EncryptedAssertion is required"
```

Comment:
I think it would be good to introduce a new constructor to the `SAML2Error` type for this.

Reply:
Added